core/primitives/src/runtime/parameter_table.rs

```diff
@@ -60,10 +60,54 @@ impl ParameterValue {
     }
 }
 
+fn format_number(mut n: u64) -> String {
+    let mut parts = Vec::new();
+    while n >= 1000 {
+        parts.push(format!("{:03?}", n % 1000));
+        n /= 1000;
+    }
+    parts.push(n.to_string());
+    parts.reverse();
+    parts.join("_")
+}
+
+impl core::fmt::Display for ParameterValue {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        match self {
+            ParameterValue::U64(v) => write!(f, "{}", format_number(*v)),
```

Comment:
Would it make sense to right-align our numbers? It can make comparisons easier.

I think something like this should work:

```suggestion
            ParameterValue::U64(v) => write!(f, "{:>20}", format_number(*v)),
```

Reply:
That's a good idea, done!